internal/subscriber/subscriber_test.go

```diff
@@ -104,3 +108,93 @@ func TestSubscriber_MessageFiltering(t *testing.T) {
 	}
 
 }
+
+func TestSubscriber_Run(t *testing.T) {
+	ctx := context.Background()
+	_, cancelFunc := context.WithCancel(ctx)
+	defer cancelFunc()
+	ctrl := gomock.NewController(t)
+	requestChan := make(chan *PullRequest)
+	responseChan := make(chan *metrov1.PullResponse)
+	errChan := make(chan error, 1000)
+	closeChan := make(chan struct{})
+	cs := getMockConsumer(ctx, ctrl)
+	consumer := getMockConsumerManager(ctx, ctrl, cs)
+	type fields struct {
+		subscriber *Subscriber
+	}
+	subs := &Subscriber{
+		subscription:        getMockSubscription(),
+		topic:               topicName,
+		subscriberID:        subID,
+		requestChan:         requestChan,
+		responseChan:        responseChan,
+		ackChan:             make(chan *AckMessage, 10),
+		modAckChan:          make(chan *ModAckMessage, 10),
+		deadlineTicker:      time.NewTicker(1 * time.Second),
+		healthMonitorTicker: time.NewTicker(500 * time.Millisecond),
+		errChan:             errChan,
+		closeChan:           closeChan,
+		consumer:            consumer,
+		cancelFunc:          cancelFunc,
+		ctx:                 ctx,
+		retrier:             &retry.Retrier{},
+		subscriberImpl:      getMockBasicImplementation(ctx, consumer, ctrl),
+	}
+	type args struct {
+		ctx context.Context
+	}
+	tests := []struct {
+		name             string
+		fields           fields
+		args             args
+		maxNumOfMessages int32
+		expectedMsg      []string
+	}{
+		{
+			name: "Test Subscriber Run",
+			fields: fields{
+				subscriber: subs,
+			},
+			maxNumOfMessages: 1,
+			expectedMsg:      []string{"a", "b"},
+			args:             args{ctx: ctx},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cs.EXPECT().ReceiveMessages(gomock.Any(), messagebroker.GetMessagesFromTopicRequest{NumOfMessages: tt.maxNumOfMessages, TimeoutMs: 1000}).Return(
+				&messagebroker.GetMessagesFromTopicResponse{
+					Messages: getMockReceivedMessages(tt.expectedMsg),
+				}, nil)
+			cs.EXPECT().FetchConsumerLag(gomock.Any()).Return(map[string]uint64{
+				"primary-topic=0": 0,
+				"retry-topic=0":   0,
+			}, nil).AnyTimes()
+			sub := tt.fields.subscriber
+			go sub.Run(tt.args.ctx)
+			<-time.NewTicker(time.Nanosecond * 1000).C
```

Comment:
why we have this ticker..?

Reply:
I am making sure that subscriber is started fine,  then sending message to request to requestChan

Reply:
you can remove this ticker, as the select block will not exit till one of the defined case is true.

Reply:
removed